src/common/services/api/cart.service.js

```diff
@@ -67,64 +67,73 @@ class Cart {
+
+      let designationType
+      angular.forEach(item.itemDefinition['details'], (v, k) => {
+        if (v['name'] === 'designation_type') {
+          designationType = v['display-value']
+        }
+      })
 
-          return {
-            uri: item.self.uri,
-            code: item.itemCode.code,
-            displayName: item.itemDefinition['display-name'],
-            designationType: designationType,
-            price: item.rate.cost.display,
-            priceWithFees: item.rate.cost['display-with-fees'],
-            config: itemConfig,
-            frequency: frequency,
-            amount: item.rate.cost.amount,
-            amountWithFees: item.rate.cost['amount-with-fees'],
-            designationNumber: item.itemCode['product-code'],
-            productUri: item.item.self.uri,
-            giftStartDate: giftStartDate,
-            giftStartDateDaysFromNow: giftStartDateDaysFromNow,
-            giftStartDateWarning: giftStartDateDaysFromNow >= 275
-          }
-        })
+      return {
+        uri: item.self.uri,
+        code: item.itemCode.code,
+        displayName: item.itemDefinition['display-name'],
+        designationType: designationType,
+        price: item.rate.cost.display,
+        priceWithFees: item.rate.cost['display-with-fees'],
+        config: itemConfig,
+        frequency: frequency,
+        amount: item.rate.cost.amount,
+        amountWithFees: item.rate.cost['amount-with-fees'],
+        designationNumber: item.itemCode['product-code'],
+        productUri: item.item.self.uri,
+        giftStartDate: giftStartDate,
+        giftStartDateDaysFromNow: giftStartDateDaysFromNow,
+        giftStartDateWarning: giftStartDateDaysFromNow >= 275
+      }
+    })
 
-        let cartTotal
-        const frequencyTotals = map(cartResponse.rateTotals, rateTotal => {
-          if (rateTotal.recurrence.interval === 'NA') {
-            cartTotal = rateTotal.cost.amount
-          }
-          return {
-            frequency: rateTotal.recurrence.display,
-            amount: rateTotal.cost.amount,
-            amountWithFees: rateTotal.cost['amount-with-fees'],
-            total: rateTotal.cost.display,
-            totalWithFees: rateTotal.cost['display-with-fees']
-          }
-        })
+    let cartTotal
+    let cartTotalDisplay
+    const frequencyTotals = map(cartResponse.rateTotals, rateTotal => {
+      if (rateTotal.recurrence.interval === 'NA') {
+        cartTotal = rateTotal.cost.amount
+        cartTotalDisplay = rateTotal.cost.display
+      }
+      return {
+        frequency: rateTotal.recurrence.display,
+        amount: rateTotal.cost.amount,
+        amountWithFees: rateTotal.cost['amount-with-fees'],
+        total: rateTotal.cost.display,
+        totalWithFees: rateTotal.cost['display-with-fees']
+      }
+    })
+    if (cartTotal) {
+      frequencyTotals.unshift(frequencyTotals.pop())
+    }
```

Comment:
Could you explain the need for this logic?

Reply:
Yeah, so the display is expecting a specific order, with `Single` gifts rate total being first. The API is returning `Single` gifts rate total last. Maybe there's a better way to do it.

Reply:
I can put a comment in the code as well.

Reply:
Ah okay, that makes sense. Yeah, I think a comment would be good.